### PR TITLE
add PipelineEndpoint demo notebook

### PR DESCRIPTION
**Pull request by Wuliyuanulb**

### CODE REVIEW

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azureml-core 0.1.0.0 (c:\\users\\liyw\\pycharmprojects\\azuremlcli\\src\\azureml-core), Requirement.parse('azureml-core~=1.9.0')).\n",
```
Comment (wangchao1230) [inline samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: remove these warnings

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+     "output_type": "stream",
+     "text": [
+      "Link to Azure Machine Learning Portal: https://ml.azure.com/experiments/TestPipelineEndpoint/runs/e29f78ab-21d3-41d6-a2da-c7cf207b66e7?wsid=/subscriptions/4faaaf21-663f-4391-96fd-47197c630979/resourcegroups/DesignerTestRG/workspaces/DesignerTest-EUS\n",
+      "Using existing pipeline endpoint \"test_pipeline_endpoint\"\n"
+     ]
+    }
+   ],
+   "source": [
+    "pipeline_parameter = {'_select_columns': select_columns}\n",
+    "pipeline = test_pipeline(**pipeline_parameter)\n",
```
Comment (wangchao1230) [inline samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: why not (_select_columns = select_columns)

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+   "source": [
+    "experiment_name = \"TestPipelineEndpoint\"\n",
+    "module_func = Module.load(workspace, namespace='azureml', name='Select Columns in Dataset')\n",
+    "test_pipeline_endpoint_name = \"test_pipeline_endpoint\"\n",
+    "test_pipeline_endpoint_id = \"b27eb28c-da34-4064-9c13-e515fdb2c71b\"\n",
+    "\n",
+    "blob_input_data = load_or_register_dataset(workspace, name='Automobile_price_data_(Raw)',\n",
+    "                                           datastore_name='azureml_globaldatasets',\n",
+    "                                           path='GenericCSV/Automobile_price_data_(Raw)')\n",
+    "select_columns = \"{\\\"isFilter\\\":true,\\\"rules\\\":[{\\\"exclude\\\":false,\\\"ruleType\\\":\\\"AllColumns\\\"},\" \\\n",
```
Comment (wangchao1230) [inline samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: I think they is a simpler form of this paramter.

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+      "Link to Azure Machine Learning Portal: https://ml.azure.com/experiments/TestPipelineEndpoint/runs/e29f78ab-21d3-41d6-a2da-c7cf207b66e7?wsid=/subscriptions/4faaaf21-663f-4391-96fd-47197c630979/resourcegroups/DesignerTestRG/workspaces/DesignerTest-EUS\n",
+      "Using existing pipeline endpoint \"test_pipeline_endpoint\"\n"
+     ]
+    }
+   ],
+   "source": [
+    "pipeline_parameter = {'_select_columns': select_columns}\n",
+    "pipeline = test_pipeline(**pipeline_parameter)\n",
+    "run = pipeline.submit(experiment_name=experiment_name)\n",
+    "published_pipeline = run.publish_to_endpoint(pipeline_endpoint_name=test_pipeline_endpoint_name,\n",
```
Comment (wangchao1230) [inline samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: update this to new pipelineEndpoint.publish api

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb (hunk 2 above)
Comment (Wuliyuanulb) [reply samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: Changed. thanks

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb (hunk 4 above)
Comment (Wuliyuanulb) [reply samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: updated.
Added publish test at the beginning

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb (hunk 3 above)
Comment (Wuliyuanulb) [reply samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: changed to ```
select_columns = "{\"isFilter\":true,\"rules\":[{\"exclude\":true,\"ruleType\":\"ColumnNames\",\"columns\":" \
                 "[\"normalized-losses\"]}]}"
```

File: samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb (hunk 1 above)
Comment (Wuliyuanulb) [reply samples_pipelines/how-to-use/pipeline_endpoint_demo.ipynb]: Removed.

File: samples_pipelines/how-to-use/sample_pipeline_endpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "pipeline_default.id: 19d79c94-279d-486f-b7f8-5014bd5e2071\n",
+      "pipeline_endpoint default version: 19d79c94-279d-486f-b7f8-5014bd5e2071\n"
+     ]
+    }
+   ],
+   "source": [
+    "# no vwesion is specified, got the default version\n",
```
Comment (wangchao1230) [inline samples_pipelines/how-to-use/sample_pipeline_endpoint.ipynb]: typo vwesion

Comment (Wuliyuanulb) [reply samples_pipelines/how-to-use/sample_pipeline_endpoint.ipynb]: Corrected thanks

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**